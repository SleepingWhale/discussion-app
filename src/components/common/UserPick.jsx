import React from 'react';
import PropTypes from 'prop-types';
import styles from './UserPick.module.css';

export function UserPick({ url, title }) {
  return url ? (
    <img src={url} className={styles.img} alt="u" title={title} />
  ) : (
    <div className={styles.placeHolder} title={title}>
      <i className="material-icons md-24 md-light">sentiment_very_satisfied</i>
    </div>
  );
}

UserPick.propTypes = {
  url: PropTypes.string,
  title: PropTypes.string
};

UserPick.defaultProps = {
  url: null,
  title: ''
};
