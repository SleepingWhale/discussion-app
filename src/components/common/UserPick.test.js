import React from 'react';
import { shallow } from 'enzyme';
import { UserPick } from './UserPick';

const url = '/some/path';
const title = 'hello world';

describe('UserPick', () => {
  it('should render img correctly', () => {
    const component = shallow(<UserPick url={url} title={title} />);
    expect(component).toMatchSnapshot();
  });

  it('should render placeholder correctly', () => {
    const component = shallow(<UserPick />);
    expect(component).toMatchSnapshot();
  });
});
