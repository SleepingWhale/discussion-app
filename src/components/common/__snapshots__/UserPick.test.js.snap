// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserPick should render img correctly 1`] = `
<img
  alt="u"
  className="img"
  src="/some/path"
  title="hello world"
/>
`;

exports[`UserPick should render placeholder correctly 1`] = `
<div
  className="placeHolder"
  title=""
>
  <i
    className="material-icons md-24 md-light"
  >
    sentiment_very_satisfied
  </i>
</div>
`;
